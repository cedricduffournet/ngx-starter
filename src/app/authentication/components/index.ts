export * from './auth-layout/auth-layout.component';
export * from './login-form/login-form.component';
export * from './register-form/register-form.component';
export * from './reset-password-form/reset-password-form.component';
export * from './send-reset-password-form/send-reset-password-form.component';
