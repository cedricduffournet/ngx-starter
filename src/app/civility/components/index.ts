export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
