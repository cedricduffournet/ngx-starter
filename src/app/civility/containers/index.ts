export * from './civility-add-modal/civility-add-modal.component';
export * from './civility-update-modal/civility-update-modal.component';
export * from './civility-delete-modal/civility-delete-modal.component';
export * from './civility-list-view/civility-list-view.component';
