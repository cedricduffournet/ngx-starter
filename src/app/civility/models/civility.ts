import { schema } from 'normalizr';

export interface Civility {
  id: number;
  name: string;
  code: string;
}

export const civilitySchema = new schema.Entity('civilities');
