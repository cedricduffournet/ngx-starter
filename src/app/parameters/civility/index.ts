export * from './parameter-civility.component';
