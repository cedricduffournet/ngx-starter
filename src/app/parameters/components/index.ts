export * from './referenced-table-menu/referenced-table-menu.component';
