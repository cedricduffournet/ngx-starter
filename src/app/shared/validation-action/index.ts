export * from './validation-action.module';
