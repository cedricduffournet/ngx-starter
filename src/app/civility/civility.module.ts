import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { CivilityEffects } from '@app/civility/state/effects';
import {
  CivilityAddComponent,
  CivilityUpdateComponent,
  CivilityDeleteComponent,
  CivilityFormComponent,
  CivilityItemComponent,
  CivilityItemsComponent
} from '@app/civility/components';
import {
  CivilityAddModalComponent,
  CivilityUpdateModalComponent,
  CivilityDeleteModalComponent,
  CivilityListViewComponent
} from '@app/civility/containers';
import { reducers } from '@app/civility/state/reducers';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    EffectsModule.forFeature([CivilityEffects]),
    StoreModule.forFeature('civilities', reducers)
  ],
  declarations: [
    CivilityListViewComponent,
    CivilityAddComponent,
    CivilityUpdateComponent,
    CivilityDeleteComponent,
    CivilityFormComponent,
    CivilityItemComponent,
    CivilityItemsComponent,
    CivilityAddModalComponent,
    CivilityUpdateModalComponent,
    CivilityDeleteModalComponent
  ],
  entryComponents: [
    CivilityUpdateModalComponent,
    CivilityAddModalComponent,
    CivilityDeleteModalComponent
  ],
  exports: [CivilityListViewComponent]
})
export class CivilityModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
