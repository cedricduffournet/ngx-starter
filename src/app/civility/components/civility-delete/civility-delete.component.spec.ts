import { TestBed, ComponentFixture } from '@angular/core/testing';

import { TranslateModule } from '@ngx-translate/core';
import { ValidationActionModule } from '@app/shared/validation-action';

import { CivilityDeleteComponent } from '@app/civility/components';
import { Civility } from '@app/civility/models/civility';

describe('CivilityDelete', () => {
  let fixture: ComponentFixture<CivilityDeleteComponent>;
  let component: CivilityDeleteComponent;
  const civility: Civility = {
    id: 1,
    code: 'testcode1',
    name: 'testname1'
  };

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [TranslateModule.forRoot(), ValidationActionModule],
      declarations: [CivilityDeleteComponent]
    });

    fixture = TestBed.createComponent(CivilityDeleteComponent);
    component = fixture.componentInstance;
    component.civility = civility;
  });

  it('should be created', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit delete event', () => {
    spyOn(component.delete, 'emit');
    component.onDelete(civility);
    expect(component.delete.emit).toHaveBeenCalledWith(civility);
  });

  it('should emit cancel event', () => {
    spyOn(component.cancel, 'emit');
    component.onCancel();
    expect(component.cancel.emit).toHaveBeenCalled();
  });

  it('should pass deleting to component', () => {
    component.deleting = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
