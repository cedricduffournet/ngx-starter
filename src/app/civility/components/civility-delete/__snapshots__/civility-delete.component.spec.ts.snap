// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CivilityDelete should be created 1`] = `
<app-civility-delete
  cancel={[Function EventEmitter]}
  civility={[Function Object]}
  delete={[Function EventEmitter]}
  deleting="false"
>
  <section
    class="bg-white"
  >
    <div
      class="p-3"
    >
       QUESTION_DELETE CIVILITY_DELETE 
    </div>
  </section><app-wrapper-validation-action>
    <div
      class="d-flex justify-content-end align-items-center p-3"
    >
      <app-validation-button
        classconfirm="danger"
        ng-reflect-class-confirm="danger"
        ng-reflect-confirm-disabled="false"
        ng-reflect-label-cancel="CANCEL"
        ng-reflect-label-confirm="DELETE"
      >
        <div
          class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
          ng-reflect-klass="button-validation-action text-"
          ng-reflect-ng-class=""
        >
          
          <button
            class="btn btn-cancel btn-lg btn-light"
            ng-reflect-klass="btn btn-cancel btn-lg"
            ng-reflect-ng-class="btn-light"
            type="button"
          >
             CANCEL 
          </button>
          
          <button
            class="btn btn-confirm btn-lg btn-danger"
            ng-reflect-klass="btn btn-confirm btn-lg"
            ng-reflect-ng-class="btn-danger"
            type="submit"
          >
            
            
            
            <span>
              DELETE
            </span>
          </button>
        </div>
      </app-validation-button>
    </div>
  </app-wrapper-validation-action>
</app-civility-delete>
`;

exports[`CivilityDelete should pass deleting to component 1`] = `
<app-civility-delete
  cancel={[Function EventEmitter]}
  civility={[Function Object]}
  delete={[Function EventEmitter]}
  deleting={[Function Boolean]}
>
  <section
    class="bg-white"
  >
    <div
      class="p-3"
    >
       QUESTION_DELETE CIVILITY_DELETE 
    </div>
  </section><app-wrapper-validation-action>
    <div
      class="d-flex justify-content-end align-items-center p-3"
    >
      <app-validation-button
        classconfirm="danger"
        ng-reflect-class-confirm="danger"
        ng-reflect-confirm-disabled="true"
        ng-reflect-label-cancel="CANCEL"
        ng-reflect-label-confirm="DELETE"
      >
        <div
          class="button-validation-action text-center d-flex justify-content-end align-items-center p-3-group"
          ng-reflect-klass="button-validation-action text-"
          ng-reflect-ng-class=""
        >
          
          
          <button
            class="btn btn-confirm btn-lg btn-danger"
            disabled=""
            ng-reflect-klass="btn btn-confirm btn-lg"
            ng-reflect-ng-class="btn-danger"
            type="submit"
          >
            
            <span
              class="spinner-border spinner-border-sm align-baseline"
              style="font-size: 1rem;"
            />
            
            <span>
              Â LOADING
            </span>
            
          </button>
        </div>
      </app-validation-button>
    </div>
  </app-wrapper-validation-action>
</app-civility-delete>
`;
